scio-core/src/main/scala/com/spotify/scio/bigquery/dynamic/package.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2016 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.bigquery
+
+import com.spotify.scio.bigquery.types.BigQueryType.HasAnnotation
+import com.spotify.scio.io.Tap
+import com.spotify.scio.testing.TypedBigQueryIO
+import com.spotify.scio.util.Functions
+import com.spotify.scio.values.SCollection
+import org.apache.beam.sdk.io.gcp.{bigquery => bqio}
+import org.apache.beam.sdk.io.gcp.bigquery.BigQueryIO.Write.{CreateDisposition, WriteDisposition}
+import org.apache.beam.sdk.io.gcp.bigquery.{DynamicDestinations, TableDestination}
+import org.apache.beam.sdk.values.ValueInSingleWindow
+
+import scala.concurrent.Future
+import scala.reflect.ClassTag
+import scala.reflect.runtime.universe._
+
+package object dynamic {
+
+  implicit class DynamicBigQuerySCollection[T](val self: SCollection[T]) extends AnyVal {
+
+    def saveAsBigQuery(name: String,
```

Comment:
Instead of `formatFn`, remove that and add a `implicit ev T <:< TableRow`? That's more consistent with vanilla `saveAsBigQuery` right?

Reply:
@nevillelyh removing the type constraint in favour of a function ... removes the need for `.toTableRow` and `.fromTableRow` map steps when using `saveAsTypedBigQuery`.

Since this is opt-in and only used internally as underlying implementation i thought it was ok. 

If you still think it's better to remove it let me know.